rust-symcrypt/src/ecc/mod.rs

```diff
@@ -382,12 +397,12 @@ impl EcKey {
     }
 
     /// `get_curve_size()` returns a `u32` that represents the size of the curve associated with the [`EcKey`].
-    /// This call cannot fail.
-    pub fn get_curve_size(&self) -> u32 {
+    /// This call can fail with `SymCryptError::MemoryAllocationFailure` if there is not enough memory to allocate the curve.
+    pub fn get_curve_size(&self) -> Result<u32, SymCryptError> {
         unsafe {
-            let ec_curve = InnerEcCurve::new(self.get_curve_type());
+            let ec_curve = InnerEcCurve::new(self.get_curve_type())?;
```

Comment:
Hm, you have `CurveType::get_size` now - why not just call that instead of allocating a new curve object just to get the size?

Reply:
Changed to return the size from the `CurveType` impl. 

` pub fn get_curve_size(&self) -> u32 {  ... ` as this call can no longer fail